### ISSUE TITLE
(Self-Voicing) Option for pink focus outline on mouse over (mouse hover)

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab-basics#221]
@jessegreenberg is it possible to create an option that can turn on the pink focus outline for interactive objects that do not have an already designed in hover style?

For GFLB, this means would it be possible to have an option to turn on the pink focus outline when a learner mouses over and interacts with either of the mass spheres and any of the checkboxes. 

The mass controls and the reset all button already have hover styles.

The reason I am asking is that for some learners, especially this with low vision seeing where their attention is helpful. The change in cursor may not be available if they have customized their cursor to be something different than the default or have changed the size of their cursor.

I am asking for this to be an option that would be available in the option menu. 

If this is possible, it would be a checkbox in the options menu, and possible labels could be:
- Hover shown on all interactions
- Hover style on all interactions
- Mouse hover shown on all interactions
- Mouse focus should on all interactions

**Comment by terracoda** [comment]
@jessegreenberg, I should first ask if it is possible to turn on the pink focus outline for mouse hover? This would be an opt-in visual option. It was suggested during a recent interview with some one who has low vision.

Some interactive controls, like the number spinners in GFLB already have a hover style, so I was firs thinking that it made sense to add a the pink outline to interactive objects that don't already have an explicit hover style, but maybe it would be good to just add pink highlight to all interactive elements and see how that works for folks.

**Comment by jessegreenberg** [comment]
@terracoda yes this is possible. I agree it would be best in the long run if we can do this for all interactive elements, rather than those that do not have a hover style already.

Would the option be for prototyping or is this something that would be in the final version?

**Comment by jessegreenberg** [comment]
Discussed briefly in meeting 8/12/19 - we talked about having the pink focus highlight surround interactive components as a potential option separate from the self-voicing feature. But for rapid prototyping, we are going to put these in the existing settings that currently live in the "Options" Dialog.

**Comment by jessegreenberg** [comment]
In the long run, I would like to think about how this feature will build into scenery and the existing FocusOverlay. But to add this quickly I am going to use the existing prototype SpeakerHighlighter and ShapeHitDetector to implement this.

**Comment by jessegreenberg** [comment]
An initial commit for this was made above. It is working pretty well, though there are a couple of issues. It is always enabled at the moment, next step will be to add a setting for it.

**Comment by jessegreenberg** [comment]
OK, the highlights are enabled through options dialog, default is off. #224 is a bigger problem now because it happens that dialog options are right on top of a readable block in the sim. Ill see about a short term fix.

**Comment by jessegreenberg** [comment]
Over slack @terracoda requested that the default be to have the highlights on.

**Comment by terracoda** [comment]
The pink hover style works great for me.
We have the known issue of the content and interactions bleeding through dialogs and menus, see #224 .
Other than that looks good to me.
